system-addon/content-src/components/TopSites/TopSiteForm.jsx

```diff
@@ -102,33 +98,28 @@ export class TopSiteForm extends React.PureComponent {
 
     return (
       <form className="topsite-form">
-        <section className="edit-topsites-inner-wrapper">
+        <div className="form-input-container">
           <div className="form-wrapper">
             <h3 className="section-title">
               <FormattedMessage id={showAsAdd ? "topsites_form_add_header" : "topsites_form_edit_header"} />
             </h3>
-            <div className="field title">
-              <input
-                type="text"
-                value={this.state.label}
-                onChange={this.onLabelChange}
-                placeholder={this.props.intl.formatMessage({id: "topsites_form_title_placeholder"})} />
-            </div>
-            <div className={`field url${this.state.validationError ? " invalid" : ""}`}>
-              <input
-                type="text"
-                ref={this.onUrlInputMount}
-                value={this.state.url}
-                onChange={this.onUrlChange}
-                placeholder={this.props.intl.formatMessage({id: "topsites_form_url_placeholder"})} />
-              {this.state.validationError &&
-                <aside className="error-tooltip">
-                  <FormattedMessage id="topsites_form_url_validation" />
-                </aside>
-              }
-            </div>
+            <TopSiteFormInput onChange={this.onLabelChange}
+              value={this.state.label}
+              titleId="topsites_form_title_label"
+              placeholderId="topsites_form_title_placeholder"
+              errorMessageId="topsites_form_url_validation"
+              {...this.props} />
+            <TopSiteFormInput onChange={this.onUrlChange}
+              value={this.state.url}
+              onClear={this.onClearUrlClick}
+              validationError={this.state.validationError}
+              titleId="topsites_form_url_label"
+              placeholderId="topsites_form_url_placeholder"
+              errorMessageId="topsites_form_url_validation"
+              {...this.props} />
           </div>
-        </section>
+          <TopSiteLink link={this.props.site || {}} title={this.state.label} preventNavigation={true} />
```

Comment:
Why is this inside the container div instead of the form-wrapper? In particular, is it expected that the `<h3>` title wraps at the same width as the form inputs? E.g.,
```
Title title
title
[ input    ] [link]
```

vs
```
Title title title
[ input    ] [link]
```

Reply:
(Maybe `h3` should be moved out like link?)

Reply:
I ended up wrapping the inputs and preview together in a div to be it easier to reason about the position of the preview in relation to the first input field. It should be (16px for the text + 8px margin) from the top.
And move the title outside so that it can match your 2nd example.